server/game/cards/06-CotE/StayYourHand.js

```diff
@@ -0,0 +1,22 @@
+const _ = require('underscore');
+const DrawCard = require('../../drawcard.js');
+
+class StayYourHand extends DrawCard {
+    setupCardAbilities() {
+        this.wouldInterrupt({
+            title: 'Cancel a duel',
+            when: {
+                onDuelInitiated: (event, context) =>
+                    event.context.player === context.player.opponent &&
+                    _.some(event.context.targets, (card) => card.controller === context.player)
```

Comment:
The second part of this condition isn't problematic I don't think, but if it's ever needed something has gone horribly wrong.... :/

Reply:
Agreed.  I was just following the exact definition of the card.  They could possibly print a duel that only targets your own characters at some point?